frontend/src/app/reducers/varianttests.js

```diff
@@ -0,0 +1,46 @@
+
+import seedrandom from 'seedrandom';
+
+import { handleActions } from 'redux-actions';
+
+function random(choices) {
+  let summedWeight = 0;
+  const variants = [];
+  Object.entries(choices).forEach(([name, weight]) => {
+    summedWeight += weight;
+    for (let i = 0; i < weight; i++) {
+      variants.push(name);
+    }
+  });
+  let seed = window.localStorage.getItem('testpilot-varianttests-id', null);
+  if (seed === null) {
+    seed = String(Math.random());
+    window.localStorage.setItem('testpilot-varianttests-id', seed);
+  }
+  return variants[Math.floor(seedrandom(seed)() * summedWeight)];
+}
+
+const tests = [
+  {
+    name: 'installButtonBorder',
+    getValue: function getValue() {
+      if (!window.navigator.language.startsWith('en')) {
+        return false;  // User gets whatever the DefaultCase is.
+      }
+      return random({
+        bigBorder: 1,
+        default: 1
```

Comment:
The random function adds up all the weights and uses that sum as 100%, so it could be 50 and 50, but it could't be 0.5 and 0.5 because it expects to see integer weights. It can also be 1 1 2 or 3 5 7 9 or something weird.

I don't mind changing the way this works, but I copied the random function from the add-on code that implements selecting variants for Experiments, so if we change the way it works on the website we should change the way it works in the add-on as well.